slang-antlr/src/main/java/com/sonarsource/slang/parser/SLangConverter.java

```diff
@@ -252,7 +252,18 @@ public Tree visitMatchExpression(SLangParser.MatchExpressionContext ctx) {
       for (SLangParser.MatchCaseContext matchCaseContext : ctx.matchCase()) {
         cases.add((MatchCaseTree) visit(matchCaseContext));
       }
-      return new MatchTreeImpl(meta(ctx), visit(ctx.statementOrExpression()), cases);
+      TreeMetaData meta = meta(ctx);
+      return new MatchTreeImpl(
+        meta,
+        visit(ctx.statementOrExpression()),
+        cases,
+        getMatchKeyword(ctx.MATCH().getSymbol()));
+    }
+
+    private static com.sonarsource.slang.api.Token getMatchKeyword(Token matchToken) {
+      TokenLocation location = new TokenLocation(matchToken.getLine(), matchToken.getCharPositionInLine(), matchToken.getText());
+      TextRange textRange = new TextRangeImpl(location.startLine(), location.startLineOffset(), location.endLine(), location.endLineOffset());
```

Comment:
These 2 lines could be put in a helper function since it is used the same way on line 98-99

Reply:
Indeed, I created a method `toSlangToken` as well in another PR (for If-then-else keywords)
https://github.com/SonarSource/slang/pull/86/files#diff-44906d002be9e3d0bff18e98b3b954c3R270
I will copy it here as well